js/navbar/tabBar.js

```diff
@@ -225,4 +225,18 @@ if (window.platformType === 'mac') {
   })
 }
 
+tabBar.container.addEventListener('dragover', e => e.preventDefault())
+
+tabBar.container.addEventListener('drop', e => {
+  e.preventDefault()
+  var data = e.dataTransfer
+  var tab = tabs.add({
+    url: data.files[0] ? 'file://' + data.files[0].path : data.getData('text'),
+    private: tabs.get(tabs.getSelected()).private
+  })
+
+  tabBar.addTab(tab)
+  webviews.add(tab)
```

Comment:
It would probably be better to do this:

```
browserUI.addTab(tabs.add(...), {enterEditMode: false, openInBackground: !settings.get('openTabsInForeground')})
```

So that the other things that addTab does (replacing an existing empty tab if there is one, scrolling the tab into view) work as expected.

Doing that normally would create a circular dependency between tabBar and browserUI, which browserify doesn't like, but I think you can fix it by doing this:

```
require('browserUI.js').addTab(...
```
instead.

Another example of this: https://github.com/minbrowser/min/blob/master/js/navbar/contentBlockingToggle.js#L86